src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -206,16 +207,44 @@ internal static IEnumerable<DownloadDependency> GetPackageDownloads(IMSBuildProj
             }
         }
 
+        /// <summary>
+        /// Gets the centrally defined package version information.
+        /// </summary>
+        /// <param name="project">The <see cref="ProjectInstance" /> to get PackageVersion for.</param>
+        /// <param name="skippedPackageIds">An <see cref="IEnumerable{string}" /> of package ids that were already added as package references. These will be excluded from the resulted list.</param>
+        /// <returns>An <see cref="IEnumerable{CentralVersionDependency}" /> containing the package versions for the specified project.</returns>
+        internal static IEnumerable<CentralVersionDependency> GetCentralPackageVersions(IMSBuildProject project, IEnumerable<string> skippedPackageIds)
+        {
+            var skippedPackageIdsHashSet = skippedPackageIds.Select(id=>id.ToLowerInvariant()).ToHashSet();
```

Comment:
I wish the DG spec had more built in dictionaries.  Creating this hashset for every single project/TFM seems like its going to have bad performance.